src/problems/chinese_ring_puzzle.clj

```diff
@@ -0,0 +1,42 @@
+; we used the two rules of chinese ring puzzle:
+; 1) the first ring, which is hooked in the very front of the loop, can be taken off or put back on the loop any time;
+; 2) the only other ring that can be taken off from or put back on the loop is the ring next to the first ring.
+(ns problems.chinese-ring-puzzle)
+
+(defrecord State [leadRing Rings])
+
+(defn children [state]
+  (let [len (count (:Rings state)),
+        leadRing (:leadRing state),
+        Rings (:Rings state),
+        nextRing (+ leadRing 1),
+        R1 (nth Rings 0),
+        R2 (nth Rings 1),
+        R3 (nth Rings 2),
+        R4 (nth Rings 3),
+        R5 (nth Rings 4),
+        R6 (nth Rings 5),
+        R7 (nth Rings 6),
+        R8 (nth Rings 7),
+        R9 (nth Rings 8)]
+
+
+    (let [children (if (= R1 "on")
+        (let [Rings1 ["off" R2 R3 R4 R5 R6 R7 R8 R9]
```

Comment:
You really didn't need to create `R1`, `R2`, etc. (which would have shortened things a fair bit). You could just have `(cons "off" (rest Rings))` instead of this naming all the rings like you did. Not only is your way wordier, it's also a lot riskier; it would be easy to have a typo (swap `R4` and `R5`) that would be really hard to see/debug.